src/database/import.ts

```diff
@@ -115,19 +127,20 @@ export default class DatabaseImporter {
           )
         )
         .pipe(readChunks)
-        .pipe(writeChunks)
+        .pipe(new BatchChunks(payloadSize))
+        .pipe(writeBatch)
         .on("data", (res: ClientResponse<JsonType>) => responses.push(res))
         .on("error", reject)
         .once("end", () => resolve(responses));
     });
   }
 
-  private writeChunk(chunk: Data): Promise<ClientResponse<JsonType>> {
+  private doWriteBatch(batch: Batch): Promise<ClientResponse<JsonType>> {
     const doRequest = (): Promise<ClientResponse<JsonType>> => {
       return this.client.request({
-        method: "PUT",
-        path: chunk.pathname + ".json",
-        body: JSON.stringify(chunk.json),
+        method: "PATCH",
+        path: "/.json",
```

Comment:
If there is a long common ancestor, we should extract that into `path`, to reduce bytes in the payload.

I think the previous abstraction `chunk: Data` is still good.

Reply:
The casework is a bit gnarly... For example, we can PATCH the payload `{"a": "foo"}` to `/.json` but we can't PATCH `"foo"` to `/a.json`. Also can't PATCH an array (have to convert to object).

Reply:
Gave it a shot anyway, lmk what you think.